Per Channel in brack-propagation function

kwanghoon-meta opened the pull request:
Summary:
Supporting Per Channel quantization in the gradient computation function.

One workaround that I have added here is
Current QNNPACK is not designed to process [transposed weight](https://fb.workplace.com/groups/pytorch.edge.users/permalink/1283737025829921/)
Here we are simply replacing Per Channel to Per Tensor to compute a gradient (Some slow learning curve or WER degradation might be expected - We don't know, nothing is guaranteed)

Test Plan:
You can create your own synthetic model,
FP32 layer -> INT8 layer with Per Channel and see if loss is decreasing

Differential Revision: D43898794



cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

kit1980 commented:
To resolve the CLA issue, make sure to add the email address of the commit to your GitHub profile
(to see the actual email address, add .patch at the end of this PR page's URL).

kwanghoon-meta committed:
Per Channel in brack-propagation function (#97475)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/97475

Supporting Per Channel quantization in the gradient computation function.

One workaround that I have added here is
Current QNNPACK is not designed to process [transposed weight](https://fb.workplace.com/groups/pytorch.edge.users/permalink/1283737025829921/)
Here we are simply replacing Per Channel to Per Tensor to compute a gradient (Some slow learning curve or WER degradation might be expected - We don't know, nothing is guaranteed)

bypass-github-export-checks

Test Plan:
You can create your own synthetic model,
FP32 layer -> INT8 layer with Per Channel and see if loss is decreasing

Reviewed By: kimishpatel

Differential Revision: D43898794

fbshipit-source-id: 86e0007bc2f5ff97e7841a182aebb113ac096ea5

weiwangmeta reviewed (approved):
Approve to unblock bot's merge.
Do NOT manually merge this.

seemethere commented:
@pytorchbot merge -f 'Landed internally'

(Initiating merge automatically since Phabricator Diff has merged, using force because this PR might not pass merge_rules.json but landed internally)
